src/sentry/static/sentry/app/components/versionHoverCard.jsx

```diff
@@ -212,25 +213,27 @@ const VersionHoverCard = React.createClass({
               })}
             </div>}
         </div>
-      </div>
-    );
+      )
+    };
   },
 
   render() {
-    let {visible} = this.state;
+    let {loading, error, hasRepos} = this.state;
+    let header = null;
+    let body = loading
+      ? <LoadingIndicator mini={true} />
+      : error ? <LoadingError /> : null;
+
+    if (!loading && !error) {
+      let renderObj = hasRepos ? this.getBody() : this.getRepoLink();
+      header = renderObj.header;
+      body = renderObj.body;
+    }
+
     return (
-      <span onMouseEnter={this.toggleHovercard} onMouseLeave={this.toggleHovercard}>
+      <Hovercard header={header} body={body}>
```

Comment:
I can't remember if you offered to do this and I declined because I wasn't sure what you were talking about lol, but in future PRs it'd be nice to do this kind of refactoring of existing components after the fact to keep the PR smaller.

Reply:
Yeah sorry :(